crates/store/src/genesis/config/mod.rs

```diff
@@ -0,0 +1,325 @@
+
+            // It's similar to `fn create_basic_fungible_faucet`, but we need to cover more cases
+            let (mut faucet_account, faucet_account_seed) = AccountBuilder::new(init_seed)
+                .account_type(account_type)
+                .storage_mode(account_storage_mode)
+                .with_auth_component(auth)
+                .with_component(component)
+                .build()?;
+
+            faucet_account.apply_delta(&AccountDelta::new(
+                faucet_account.id(),
+                AccountStorageDelta::new(),
+                AccountVaultDelta::default(),
+                ONE,
+            )?)?;
+
+            if faucets.insert(symbol.clone(), faucet_account.id()).is_some() {
+                return Err(GenesisConfigError::DuplicateFaucetDefinition { symbol: token_symbol });
+            }
+
+            secrets.push((
+                format!("faucet_{symbol}.mac", symbol = symbol.to_lowercase()),
+                faucet_account.clone(),
+                secret_key,
+                faucet_account_seed,
+            ));
+
+            all_accounts.push(faucet_account);
+        }
+
+        let zero_padding_width = usize::ilog10(std::cmp::max(10, wallet_configs.len())) as usize;
+
+        // then setup all wallet accounts, which reference the faucet's for their provided assets
+        for (index, WalletConfig { has_updatable_code, storage_mode, assets }) in
+            wallet_configs.into_iter().enumerate()
+        {
+            let mut rng = ChaCha20Rng::from_seed(rand::random());
+            let secret_key = SecretKey::with_rng(&mut get_rpo_random_coin(&mut rng));
+            let auth = AuthScheme::RpoFalcon512 { pub_key: secret_key.public_key() };
+            let init_seed: [u8; 32] = rng.random();
+
+            let assets = Result::<Vec<_>, GenesisConfigError>::from_iter(assets.into_iter().map(
+                |AssetEntry { amount, symbol }: AssetEntry| {
+                    let token_symbol = TokenSymbol::new(&symbol)?;
+                    let faucet_id = faucets.get(&symbol).ok_or_else(|| {
+                        GenesisConfigError::MissingFaucetDefinition { symbol: token_symbol }
+                    })?;
+
+                    Ok(FungibleAsset::new(*faucet_id, amount)?)
+                },
+            ))?;
+            let account_type = if has_updatable_code {
+                AccountType::RegularAccountUpdatableCode
+            } else {
+                AccountType::RegularAccountImmutableCode
+            };
+            let account_storage_mode = storage_mode.into();
+            let (mut wallet_account, wallet_account_seed) =
+                create_basic_wallet(init_seed, auth, account_type, account_storage_mode)?;
+            wallet_account.apply_delta(&AccountDelta::new(
+                wallet_account.id(),
+                AccountStorageDelta::new(),
+                AccountVaultDelta::default(),
+                ONE,
+            )?)?;
```

Comment:
Since we are incrementing the nonce below (line 201), we don't need to increment it here.